test/e2e/proxy_test.go

```diff
@@ -55,13 +57,18 @@ type patchStringValue struct {
 	Value string `json:"value"`
 }
 
-func (u *proxyUser) shareSpaceWith(t *testing.T, hostAwait *wait.HostAwaitility, guestUser *proxyUser) {
+func (u *proxyUser) shareSpaceWith(t *testing.T, awaitilities wait.Awaitilities, guestUser *proxyUser) *toolchainv1alpha1.SpaceBindingRequest {
 	// share primaryUser space with guestUser
-	guestUserMur, err := hostAwait.GetMasterUserRecord(guestUser.compliantUsername)
+	guestUserMur, err := awaitilities.Host().GetMasterUserRecord(guestUser.compliantUsername)
 	require.NoError(t, err)
-	primaryUserSpace, err := hostAwait.WaitForSpace(t, u.compliantUsername, wait.UntilSpaceHasAnyTargetClusterSet(), wait.UntilSpaceHasAnyTierNameSet())
+	primaryUserSpace, err := awaitilities.Host().WaitForSpace(t, u.compliantUsername, wait.UntilSpaceHasAnyTargetClusterSet(), wait.UntilSpaceHasAnyTierNameSet(), wait.UntilSpaceHasAnyProvisionedNamespaces())
 	require.NoError(t, err)
-	CreateSpaceBinding(t, hostAwait, guestUserMur, primaryUserSpace, "admin") // creating a spacebinding gives guestUser access to primaryUser's space
+	spaceBindingRequest := CreateSpaceBindingRequest(t, awaitilities, primaryUserSpace.Spec.TargetCluster,
+		WithSpecSpaceRole("admin"),
+		WithSpecMasterUserRecord(guestUserMur.GetName()),
+		WithNamespace(testsupportspace.GetDefaultNamespace(primaryUserSpace.Status.ProvisionedNamespaces)),
+	)
+	return spaceBindingRequest
```

Comment:
let's wait for the SpaceBinding to be created
```suggestion
	_, err = awaitilities.Host().WaitForSpaceBinding(t, guestUserMur.GetName(), primaryUserSpace.GetName())
	require.NoError(t, err)
	return spaceBindingRequest
```